lib/domain.js

```diff
@@ -12,26 +9,43 @@ var getRulesForTld = require('./tld-rules.js');
  * @param {string} host
  * @return {String}
  */
-module.exports = function getDomain (allRules, validHosts, host) {
-  var domain = null, hostTld, rules, rule;
+module.exports = function getDomain(allRules, validHosts, host, isHostClean) {
   var _validHosts = validHosts || [];
+  var cleanHost = cleanHostValue(host, isHostClean);
 
-  if (isValid(_validHosts, host) === false) {
+  if (isValid(_validHosts, cleanHost) === false) {
     return null;
   }
 
-  host = cleanHostValue(host);
-  hostTld = extractTldFromHost(host);
-  rules = getRulesForTld(allRules, hostTld, new Rule({"firstLevel": hostTld, "isHost": _validHosts.indexOf(hostTld) !== -1}));
-  rule = getCandidateRule(host, rules);
+  // Check if `host` ends with '.' followed by one host specified in validHosts.
+  for (var i = 0; i < validHosts.length; i++) {
```

Comment:
If we loop over the `validHosts` array, I'd rather use `validHosts.some()/find()` and interrupt the loop as soon as we find the relevant `vhost`.

Reply:
The loop is already exited early when we find a match with `return vhost`. Am I missing something here?

Reply:
True! I missed the `return` statement and assumed you were using the `vhost` afterwards.